gc/structs/ForwardedHeader.hpp

```diff
@@ -546,11 +547,13 @@ class MM_ForwardedHeader
 	 * @param[in] compressed bool describing whether object references are compressed or not
 	 */
 	MM_ForwardedHeader(omrobjectptr_t objectPtr, bool compressed)
-	: _objectPtr(objectPtr)
+	:
+#if defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS)
+	_compressObjectReferences(compressed)
+#endif /* defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS) */
+	, _objectPtr(objectPtr)
```

Comment:
I don't expect this to compile when `_compressObjectReferences` is absent.

Reply:
Yes, the first field in a class can not be optional.

Reply:
I expect re-ordering the public/private/protected sections would be ok. @dmitripivkine ?

Reply:
> I expect re-ordering the public/private/protected sections would be ok. @dmitripivkine ?

re-ordering should be ok

Reply:
> > I expect re-ordering the public/private/protected sections would be ok. @dmitripivkine ?
> 
> re-ordering should be ok

however I don't understand why it can not be private any more

Reply:
Unused private fields cause the OSX compiler to complain.